src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,28 +243,190 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
-            int portNumber = 1433;
-            string[] serverAndPortParts = fullServerName.Split(',');
 
-            if (serverAndPortParts.Length == 2)
+            const int DEFAULT_SERVER_PORT = 1433;
+
+            if (String.IsNullOrWhiteSpace(fullServerName))
             {
-                try
+                return new SNITCPHandle("localhost", DEFAULT_SERVER_PORT, timerExpire, callbackObject, parallel);
+            }
+
+            string[] serverNamePartsByComma = fullServerName.Split(',');
+            string[] serverNamePartsByFowardSlash = fullServerName.Split('\\');
+            if (serverNamePartsByComma.Length < 2 && serverNamePartsByFowardSlash.Length < 2)
+            {
+                return new SNITCPHandle(fullServerName, DEFAULT_SERVER_PORT, timerExpire, callbackObject, parallel);
+            }
+            else if (serverNamePartsByComma.Length == 2 && serverNamePartsByFowardSlash.Length < 2)
+            {
+                string hostName = serverNamePartsByComma[0];
+                int portNumber = -1;
+                if (Int32.TryParse(serverNamePartsByComma[1], out portNumber))
```

Comment:
TryParse method does not blow up even if given string is not numeric.
Since we want to propagate all the exceptions, I brought ushort.parse() it back.